frontend/compiled/cs.go

```diff
@@ -147,3 +192,61 @@ func (cs *ConstraintSystem) AddDebugInfo(errName string, i ...interface{}) int {
 func (cs *ConstraintSystem) FieldBitLen() int {
 	return cs.bitLen
 }
+
+func (cs *ConstraintSystem) GetDebugInfo() ([][]uint64, map[uint32]string) {
+	r := make([][]uint64, len(cs.DebugInfo))
+	for _, l := range cs.DebugInfo {
+		r = append(r, l.Stack)
+	}
+	return r, cs.DebugStackPaths
+}
+
+func (cs *ConstraintSystem) stack() (r []uint64) {
```

Comment:
Method very similar to `debug.WriteStack()`. Would make sense to refactor the common part into separate method and then `debug.WriteStack()` would just expand the stack entries into string builder?

Reply:
yes thought about that but the "debugStack" new version, just needs file and line number. The other debug stack for assertion failure, also collects function names. And depending on build tag, filter gnark internals or not.  

Will revisit after building the constraint viewer tool.